src/ray/object_manager/object_manager.cc

```diff
@@ -336,14 +336,15 @@ void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
     return PushLocalObject(object_id, node_id);
   }
 
-  // Avoid setting duplicated timer for the same object and node pair.
-  auto &nodes = unfulfilled_push_requests_[object_id];
-  // Issue a restore request if the object is on local disk. This is only relevant
-  // if the local filesystem storage type is being used.
+  // Push from spilled object directly if the object is on local disk.
   auto object_url = get_spilled_object_url_(object_id);
   if (!object_url.empty()) {
```

Comment:
Actually, we should also check if the external storage is a local file system, not S3. Can you add an additional check here?

Reply:
Read the code again. So if it's not local spilled file the sender will not actually sending the request over? If that's the case we probably don't need to handle it? (as anyway if we happened to receive a external url, it will fail at the sender side)

Reply:
actually, never mind. there is a case this could happens if we have race condition...

Reply:
get_spilled_object_url_ only returns if the object was spilled from the node, probably we should rename it to something with clearer semantics.

However I believe the URL could either refer to local fs storage or remote storage, we should check RayConfig::instance().is_external_storage_type_fs() to verify we're using local fs.